Updated to replace GS Collections with Eclipse Collections

mlalma opened the pull request.

jvirtanen commented:
Looking good! Just one small thing. Currently all dependencies are defined in the top-level POM, mainly in order to keep the dependency versions in different Maven modules consistent. So, it would be good if

1. Eclipse Collections would be defined in the top-level POM, and
2. the dependency version would be defined as a property (see `jmh.version` for an example).

jvirtanen commented:
(The dependency version should be defined as a property, because it is used for two dependencies.)

mlalma commented:
Added Eclipse Framework dependencies to top-level DOM, added Eclipse Framework version to properties structure on parity top-level pom.xml (eclfw.version).

Removed Eclipse Framework version from parity-client's pom.xml.

jvirtanen commented on pom.xml:
```diff
@@ -36,6 +36,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <java.version>1.8</java.version>
     <jmh.version>0.5.6</jmh.version>
+    <eclfw.version>7.0.0</eclfw.version>
```
What does this stand for? Perhaps this could be also `eclipse.collections.version` for clarity.

jvirtanen replied:
Eclipse has many frameworks but I guess this version number is unique to Eclipse Collections, right?

mlalma commented:
Good point; I don't know what I was thinking, when I renamed Eclipse Collections to Eclipse Framework. Fixed it.

jvirtanen commented:
Almost there! I noticed there are some tabs still mixed with spaces in the indentation; see the inline comments for details.

jvirtanen commented on pom.xml:
```diff
@@ -54,9 +55,14 @@
-        <version>5.1.0</version>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections-api</artifactId>
+    	<version>${eclipse.collections.version}</version>
```
Tabs used in this block.

jvirtanen commented on pom.xml:
```diff
@@ -54,9 +55,14 @@
+      <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections</artifactId>
+    	<version>${eclipse.collections.version}</version>
```
Tabs used in this block.

jvirtanen commented on parity-client/pom.xml:
```diff
@@ -53,6 +49,14 @@
     </dependency>
+    <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections-api</artifactId>
```
Tabs used in this block.

jvirtanen commented on parity-client/pom.xml:
```diff
@@ -53,6 +49,14 @@
+    </dependency>
+    <dependency>
+    	<groupId>org.eclipse.collections</groupId>
+    	<artifactId>eclipse-collections</artifactId>
```
Tabs used in this block.

jvirtanen commented:
A tip: `git diff` highlights mixed spaces and tabs prominently in its output.

mlalma commented:
Replaced tabs with spaces on pom.xml files

jvirtanen commented:
Merged, thank you!
